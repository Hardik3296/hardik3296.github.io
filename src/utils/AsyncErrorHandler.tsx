import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;