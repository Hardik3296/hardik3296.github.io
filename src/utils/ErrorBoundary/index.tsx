import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;