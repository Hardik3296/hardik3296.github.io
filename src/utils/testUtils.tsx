import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}