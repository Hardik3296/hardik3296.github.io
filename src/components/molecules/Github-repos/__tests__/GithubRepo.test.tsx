import GithubRepo from "..";
import {shallow, ShallowWrapper} from "enzyme";
import renderer from "react-test-renderer";
import {findAttributeByTestId} from "../../../../utils/testUtils";

const renderComponent=()=>{
   return shallow(<GithubRepo/>);
}

describe("Github Repos",()=>{
   let wrapper:ShallowWrapper;

   beforeEach(()=>{
      wrapper = renderComponent();
   });

   // it("Should have 6 repo",()=>{
   //    const component = findAttributeByTestId(wrapper,"repo-name");
   //    expect(component.length).toBe(6);
   // });

   it("Should render correctly",()=>{
      const component = renderer.create(<GithubRepo/>).toJSON();
      expect(component).toMatchSnapshot();
   })
})