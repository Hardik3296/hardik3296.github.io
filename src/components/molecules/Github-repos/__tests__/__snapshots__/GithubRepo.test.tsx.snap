// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github Repos Should render correctly 1`] = `
<div
  className="container [object Object]"
  id="github-repos"
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;
